index.d.ts

```diff
@@ -1,38 +1,38 @@
 // TypeScript Version: 3.7
 declare module "effection" {
-  export type Operation = OperationFn | Sequence | Promise<any> | Controller | undefined;
+  export type Operation<T = any> = undefined | OperationFn<T> | Sequence<T> | PromiseLike<T> | Controller<T>;
```

Comment:
I've defaulted these types to `any`, to maximize compatibility with existing code.